demisto_sdk/commands/split/jsonsplitter.py

```diff
@@ -0,0 +1,125 @@
+import json
+import os
+
+import click
+
+from demisto_sdk.commands.common.constants import (DASHBOARDS_DIR,
+                                                   GENERIC_MODULES_DIR,
+                                                   PACKS_DIR)
+from demisto_sdk.commands.common.tools import (get_pack_name,
+                                               is_external_repository)
+
+
+class JsonSplitter:
+    """
+    JsonSplitter is a class intended to split Generic Modules from their intrinsic dashboards
+
+    Attributes:
+        input (str): The path to the Generic module json file to split.
+        output (str): The path to a directory in which the new dashboards should be put (and the module
+                    if we create a new module file).
+        no_auto_create_dir (bool): Whether to auto create new directories in content repo.
+        no_logging (bool): Whether to print logs to the user.
+        new_module_file (bool): Whether to create a new module file.
+    """
+
+    def __init__(self, input: str, output: str, no_auto_create_dir: bool = False,
+                 no_logging: bool = False, new_module_file: bool = False):
+        self.input = input
+        self.dashboard_dir = output if output else ''
+        self.module_dir = output if output else ''
+        self.logging = not no_logging
+        self.autocreate_dir = not no_auto_create_dir
+        self.new_module_file = new_module_file
+
+        with open(self.input, 'rb') as json_file:
+            self.module_json_data = json.load(json_file)
+
+    def split_json(self):
+        if self.logging:
+            click.secho(f"Starting dashboard extraction from generic module {self.module_json_data.get('name')}",
+                        fg='bright_cyan')
+        self.create_output_dirs()
+        self.create_dashboards()
+        self.create_module()
+        return 0
+
+    def create_output_dirs(self):
+        # no output given
+        if not self.dashboard_dir:
+            # check if in content repository
+            if not is_external_repository() and self.autocreate_dir:
+                pack_name = get_pack_name(self.input)
+                if self.logging:
+                    click.echo(f"No output path given creating dirs in pack {pack_name}")
+                pack_path = os.path.join(PACKS_DIR, pack_name)
+                self.dashboard_dir = os.path.join(pack_path, DASHBOARDS_DIR)
+                self.module_dir = os.path.join(pack_path, GENERIC_MODULES_DIR)
+
+                # create the dirs, dont fail if exist
+                try:
+                    os.mkdir(self.dashboard_dir)
+                except FileExistsError:
+                    pass
+
+                try:
+                    os.mkdir(self.module_dir)
+                except FileExistsError:
+                    pass
+
```

Comment:
This can be generalized to one function of create_output_dir(dir_entity_type_name) or something like that.